extensions/wikia/ARecoveryEngine/ARecoveryModule.class.php

```diff
@@ -18,16 +15,31 @@ public function isPageFairRecoveryEnabled() {
 	/**
 	 * Checks whether SourcePoint recovery is enabled (on current wiki)
 	 *
-	 * $wgAdDriverEnableSourcePointRecovery === false; // disabled on wiki
-	 * $wgAdDriverEnableSourcePointRecovery === true; // enabled on wiki
+	 * @return bool
+	 */
+	public static function isSourcePointRecoveryEnabled() {
+		global $wgUser, $wgAdDriverEnableSourcePointRecovery;
+
+		return $wgUser->isAnon() && F::app()->checkSkin( [ 'oasis' ] ) && $wgAdDriverEnableSourcePointRecovery;
+	}
+
+	/**
+	 * Checks whether SourcePoint MMS is enabled (on current wiki)
 	 *
 	 * @return bool
 	 */
-	public function isSourcePointRecoveryEnabled() {
-		global $wgUser, $wgAdDriverEnableSourcePointRecovery, $wgAdDriverEnableSourcePointMMS;
+	public static function isSourcePointMessagingEnabled() {
+		global $wgUser, $wgAdDriverEnableSourcePointMMS;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)